### PR TITLE
Staking pool by id endpoint

### PR DESCRIPTION
**Pull request by kimpers**
**Example request:** GET /staking/pools/2
**Response:**
```json
{
    "poolId": "2",
    "stakingPool": {
        "poolId": "2",
        "operatorAddress": "0x6ecbe1db9ef729cbe972c83fb886247691fb6beb",
        "createdAt": {
            "blockNumber": 14576829,
            "txHash": "0x4d587b256b712a456cca2297d599460942444f0cf04774fe48d380b06e89bfb0"
        },
        "metaData": {
            "name": "Magical Market Making Machine (MMM)",
            "bio": "All your stake are belong to us",
            "location": "San Francisco, CA",
            "isVerified": false,
            "logoUrl": "https://github.com/0xProject/0x-staking-pool-registry/raw/master/logos/0x.png",
            "websiteUrl": "http://0x.org"
        },
        "allTimeStats": {
            "operatorRewardsPaidInEth": 1.24010006641e-7,
            "membersRewardsPaidInEth": 0.06200487931017588,
            "totalRewardsPaidInEth": 0.06200500332018252,
            "protocolFeesGeneratedInEth": 0.00975
        },
        "epochRewards": [
            {
                "epochId": 1,
                "operatorRewardsPaidInEth": 0,
                "membersRewardsPaidInEth": 0,
                "totalRewardsPaidInEth": 0
            },
            {
                "epochId": 2,
                "operatorRewardsPaidInEth": 4.70105096e-9,
                "membersRewardsPaidInEth": 0.002350520778638962,
                "totalRewardsPaidInEth": 0.002350525479689922
            },
            {
                "epochId": 3,
                "operatorRewardsPaidInEth": 0,
                "membersRewardsPaidInEth": 0,
                "totalRewardsPaidInEth": 0
            },
            {
                "epochId": 4,
                "operatorRewardsPaidInEth": 1.19308955681e-7,
                "membersRewardsPaidInEth": 0.059654358531536916,
                "totalRewardsPaidInEth": 0.0596544778404926
            },
            {
                "epochId": 5,
                "operatorRewardsPaidInEth": 0,
                "membersRewardsPaidInEth": 0,
                "totalRewardsPaidInEth": 0
            },
            {
                "epochId": 6,
                "operatorRewardsPaidInEth": 0,
                "membersRewardsPaidInEth": 0,
                "totalRewardsPaidInEth": 0
            },
            {
                "epochId": 7,
                "operatorRewardsPaidInEth": 0,
                "membersRewardsPaidInEth": 0,
                "totalRewardsPaidInEth": 0
            },
            {
                "epochId": 8,
                "operatorRewardsPaidInEth": 0,
                "membersRewardsPaidInEth": 0,
                "totalRewardsPaidInEth": 0
            },
            {
                "epochId": 9,
                "operatorRewardsPaidInEth": 0,
                "membersRewardsPaidInEth": 0,
                "totalRewardsPaidInEth": 0
            }
        ]
    }
}
```

### CODE REVIEW

File: src/handlers/staking_handlers.ts [hunk 1]
```diff
@@ -13,6 +19,25 @@ export class StakingHandlers {
         };
         res.status(HttpStatus.OK).send(response);
     }
+    public async getStakingPoolByIdAsync(_req: express.Request, res: express.Response): Promise<void> {
+        const poolId = _req.params.id;
```
Comment (fragosti) [inline src/handlers/staking_handlers.ts]: should be `req` if you're actually using it.

File: src/queries/staking_queries.ts [hunk 2]
```diff
@@ -0,0 +1,389 @@
+export const currentEpochQuery = `
```
Comment (fragosti) [inline src/queries/staking_queries.ts]: 👍

File: src/types.ts [hunk 3]
```diff
@@ -159,6 +197,10 @@ export interface AllTimeStakingStats {
     totalRewardsPaidInEth: number;
 }
 
+export interface StakingPoolResponse {
+    poolId: string;
+    stats: PoolWithHistoricalStats;
```
Comment (fragosti) [inline src/types.ts]: should we call it "pool" something? it is a pool just with more info .

**Comment by hrharder** [comment]
I'm sure I'm doing something wrong, but I seem unable to get this endpoint to work on kovan or mainnet. We (@ParadigmFoundation) operate mainnet pool ID 6, which has had it's metadata merged into [`0xProject/0x-staking-pool-registry`](https://github.com/0xProject/0x-staking-pool-registry/blob/master/staking_pools.json#L3) last week.

I was also unable to get the endpoint to work w/ chain ID 42, which I would have expected it to as you show in your example. I figured I'd let you know, but like I said, it seems most likely I'm not running something properly. The error I get over HTTP is:

```
Error: Could not find pool with pool_id 6
    at StakingDataService.<anonymous> (/Users/hen/GitHub/0x-api/lib/services/staking_data_service.js:50:23)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/hen/GitHub/0x-api/lib/services/staking_data_service.js:4:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

File: src/types.ts [hunk 4]
```diff
@@ -121,6 +136,16 @@ export interface PoolWithStats extends Pool {
     sevenDayProtocolFeesGeneratedInEth: number;
 }
 
+export interface PoolWithHistoricalStats extends Pool {
+    allTimeStats: {
+        operatorRewardInEth: number;
+        membersRewardInEth: number;
+        totalRewardInEth: number;
+        protocolFeesGeneratedInEth: number;
```
Comment (fragosti) [inline src/types.ts]: I feel like we should figure out how to make this fit in with `AllTimeStakingStats` (returned by the /stats endpoint) and `AllTimePoolRewards`. Can't they be the same thing?

**Comment by fragosti** [comment]
@hrharder I think it is because you don't have any data in your table (it can take a while to catch up) and our error middleware isn't doing a great job of catching runtime errors and presenting them well as it is supposed to. That is a separate issue that we will address but not in this PR.

**Comment by hrharder** [comment]
Thanks @fragosti, that would make sense. I'll stay out of your hair :)

File: src/types.ts (hunk 4 above)
Comment (kimpers) [reply src/types.ts]: Good point, they should be the same. My bad 😳

File: src/types.ts (hunk 3 above)
Comment (kimpers) [reply src/types.ts]: How about `stakingPool` to be consistent with `stakingPools` returned from the pools endpoint?

**Pull request merged**